registry/storage/linkedblobstore.go

```diff
@@ -74,6 +75,20 @@ func (lbs *linkedBlobStore) Put(ctx context.Context, mediaType string, p []byte)
 		return distribution.Descriptor{}, err
 	}
 
+	// Remove a tombstone if one exists for this blob.  Continue to write
+	// the blob as it may have been removed by the GC sweep process.  If
+	// deletes are not enabled, fall through
+	dgst, err := digest.FromBytes(p)
```

Comment:
Shouldn't the statter be in charge of the membership? This is somewhat how this code is structured.